flask_genshi.py

```diff
@@ -274,10 +274,6 @@ def generate_template(template=None, context=None,
         context.setdefault(key, value)
     context.setdefault('filters', filters)
     context.setdefault('tests', current_app.jinja_env.tests)
-    for key, value in filters.items():
-        context.setdefault(key, value)
-    for key, value in current_app.jinja_env.tests.items():
-        context.setdefault('is%s' % key, value)
```

Comment:
Do you know why jinja tests and filters were added to the global namespace in the first place? I don't really understand the relationship between genshi and jinja, even after reading the docs. In some places, it seems like they are contrasted as two different templating solutions, and yet in other places such as this, it looks like genshi uses portions of jinja. Is this just for "ease of use?"

I don't have a problem removing the jinja globals, I am just curious.

Reply:
I can't say for sure what was the reasoning behind it, but my guess would be the following: 
- Jinja is the default template engine shipped by Flask, if you use Flask you already have jinja installed. 
- Jinja comes with a bunch of [filters](https://jinja.palletsprojects.com/en/2.11.x/templates/#builtin-filters) and [tests](https://jinja.palletsprojects.com/en/2.11.x/templates/#builtin-tests)
- Genshi does not include such utility functions and it's non-trivial to pass new function definitions through the context globally.
- Maybe the original maintainer had some use case that was already solved by one/couple of these and instead of re-implementing it, he simply added a way to call them, since he knew everyone who installs flask-genshi has them by default, as dependency from Flask.
- And finally somewhere down the line he got tired of typing `filters.lower` and simply wanted to use `lower` and thought is was a good idea to make it globally available. Since most functions that overshadow a name behave identical such as `lower`, he did not see any harm in doing this. `max` is special with that regard since it accepts both `max(iterable_var)` and `max(var1, var2 ,var3, ...)`